Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -75,17 +76,24 @@ public SiteExtensionManager(IContinuousJobsManager continuousJobManager, ITrigge
         {
             ITracer tracer = _traceFactory.GetTracer();
             var extensions = new List<SiteExtensionInfo>();
-            SourceRepository remoteRepo = GetRemoteRepository(feedUrl);
+            IEnumerable<SourceRepository> remoteRepos = GetRemoteRepositories(feedUrl);
 
             SearchFilter filterOptions = new SearchFilter();
             filterOptions.IncludePrerelease = allowPrereleaseVersions;
 
-            IEnumerable<UIPackageMetadata> packages = null;
+            IEnumerable<UIPackageMetadata> packages = new List<UIPackageMetadata>();
 
             using (tracer.Step("Search site extensions by filter: {0}", filter))
             {
-                packages = (await remoteRepo.Search(string.IsNullOrWhiteSpace(filter) ? string.Empty : filter, filterOptions: filterOptions))
-                            .OrderByDescending(p => p.DownloadCount);
+                foreach (SourceRepository remoteRepo in remoteRepos)
+                {
+                    var foundUIPackages = await remoteRepo.Search(string.IsNullOrWhiteSpace(filter) ? string.Empty : filter, filterOptions: filterOptions);
+                    if (null != foundUIPackages)
+                    {
+                        packages = packages.Concat(foundUIPackages);
+                    }
+                }
+                packages.OrderByDescending(p => p.DownloadCount);
```

Comment:
Originally I had:
```packages.Concat(await remoteRepo.Search(string.IsNullOrWhiteSpace(filter) ? string.Empty : filter, filterOptions: filterOptions));```
but it turns out that .Concat doesn't actually concat to packages; it simply returns the concatination of packages + 1st arg.  This was the cause of the last [bug fix](https://github.com/projectkudu/kudu/pull/2568/commits/0891d0e63b546d6f4061568a9ac4c01673c61100).
From the description of [foo.Concat(bar) ](https://msdn.microsoft.com/en-us/library/bb302894(v=vs.110).aspx), I foolishly thought that it would concatenate bar to foo in-place. I was wrong; foo is not modified by .Concat().

Reply:
Noe of the Linq methods ever have side effects. It's all about building new enumerations over existing ones.